### PR TITLE
Do not do VBlankIntrWait if using the wireless adapter

### PR DESCRIPTION
**Pull request by AtariDreams**
Desynchronization can occur otherwise.

## **Discord contact info**
roseareazero

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -420,7 +420,16 @@ static void IntrDummy(void)
...
+
+    if (gWirelessCommType != 0)
+    {
+        // Desynchronization may occur if wireless adapter is connected
+        // and we call VBlankIntrWait();
+        while (!(gMain.intrCheck & INTR_FLAG_VBLANK))
+            ;
+    }
+    else
+        VBlankIntrWait();
```
Comment (LOuroboros) [inline src/main.c:432]:
```suggestion
    else
    {
        VBlankIntrWait();
    }
```

**Comment by AtariDreams** [comment]
Done!

**Comment by AlexOn1ine** [comment]
> Done!

Please don't force push after a review has happened. I personally don't see any problems with force pushing before a review but after a review it's better to just make a new commit. I know it doesn't matter much in this case because it's is a small PR but it's still good practice.

**Comment by AsparagusEduardo** [comment]
> Please don't force push after a review has happened. I personally don't see any problems with force pushing before a review but after a review it's better to just make a new commit. I know it doesn't matter much in this case because it is a small PR but it's still good practice.

The force-push was made in the initial commit, but the fix was made without force-pushing, to they did what you said.
Still unnecessary for the first one, though ^^;

**Comment by AsparagusEduardo** [comment]
Note for the reviewer:
This needs to be reviewed in-game to see if it causes performance issues compared to the current state.

**Comment by kleeenexfeu** [comment]
> Note for the reviewer: This needs to be reviewed in-game to see if it causes performance issues compared to the current state.

I've been trying it a few days, didn't notice any difference

**Comment by Bassoonian** [comment]
To which extent does this fix differ from the [tutorial from pret's wiki](https://github.com/pret/pokeemerald/wiki/Improving-the-WaitForVBlank-function) and, perhaps more importantly, how is it better? And if it is better, why isn't it the suggested solution on the pret wiki?

**Comment by AtariDreams** [comment]
> To which extent does this fix differ from the [tutorial from pret's wiki](https://github.com/pret/pokeemerald/wiki/Improving-the-WaitForVBlank-function) and, perhaps more importantly, how is it better? And if it is better, why isn't it the suggested solution on the pret wiki?

Because pret did not take the wireless adapter into account and most pret romhacks don't

**Comment by Bassoonian** [comment]
> > To which extent does this fix differ from the [tutorial from pret's wiki](https://github.com/pret/pokeemerald/wiki/Improving-the-WaitForVBlank-function) and, perhaps more importantly, how is it better? And if it is better, why isn't it the suggested solution on the pret wiki?
> 
> Because pret did not take the wireless adapter into account and most pret romhacks don't

I'm sorry, but did you read the tutorial? There's literally a block of code specifically for retaining Wireless Adapter functionality

**Comment by AtariDreams** [comment]
> > > To which extent does this fix differ from the [tutorial from pret's wiki](https://github.com/pret/pokeemerald/wiki/Improving-the-WaitForVBlank-function) and, perhaps more importantly, how is it better? And if it is better, why isn't it the suggested solution on the pret wiki?
> 
> > 
> 
> > Because pret did not take the wireless adapter into account and most pret romhacks don't
> 
> 
> 
> I'm sorry, but did you read the tutorial? There's literally a block of code specifically for retaining Wireless Adapter functionality

This PR predates that edit. Nintendo games that also use the adapter like FZero take this approach.

**Comment by Bassoonian** [comment]
Sure, it predates that edit indeed, but still doesn't answer my other questions: how is your approach better than the tutorial's and why should we use it over the tutorial's?

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**